tensorflow/python/kernel_tests/one_hot_op_test.py

```diff
@@ -34,7 +34,8 @@ def _testOneHot(self, truth, use_gpu=False, expected_err_re=None,
         ans = tf.one_hot(**inputs)
         if expected_err_re is None:
           tf_ans = ans.eval()
-          self.assertAllClose(tf_ans, truth, atol=1e-10)
+          if inputs.get('dtype') is not tf.string:
+            self.assertAllClose(tf_ans, truth, atol=1e-10)
```

Comment:
i think one_hot should probably return exact values.  have you tried converting this to self.assertAllEqual, including for strings?